packages/pds/src/services/repo/index.ts

```diff
@@ -46,10 +46,24 @@ export class RepoService {
     ])
   }
 
-  async processWrites(did: string, writes: PreparedWrite[], now: string) {
+  async processWrites(
+    did: string,
+    writes: PreparedWrite[],
+    now: string,
+    pinnedForUpdate?: PinnedSqlRepoStorage,
```

Comment:
Could we just thread the `swapCommit` through to the repo service methods?

ie
```ts
processWrites(
    did: string,
    writes: PreparedWrite[],
    now: string,
    swapCommit?: CID
)
```

Same with `swapRecord`: thread it through on `prepareUpdate`/`prepareDelete`, etc

Reply:
I think we might need to take a lock on the repo for a swap-record since racing writes on the same record could see the same current cid for the record, and then after that get linearized when the lock is taken (so one of the writes would clobber the other).

Reply:
ah that's a good point 🤔 

kk lmk how it goes. could also maybe return something from the repo service methods (like prevHead & prevRecord) so that the consuming code can throw & revert the tx if the CAS fails.

if it's too headachy don't spin on it too long. just feel like the PinnedStorage is a bit leaky of an abstraction

Reply:
I ended up putting the record cas on the prepared writes, and the commit cas as an argument to `processWrites()`.  Seemed to work well 👍